File: StudentGrade/python/student_grade_main.py
from student_grade import *

student_quiz_results = []

number_of_students = 0
while number_of_students <= 0:
    number_of_students = int(input("How many students do you have?\n"))

    if number_of_students <= 0:
        print("Invalid input")

number_of_subjects = 0
while number_of_subjects <= 0:
    number_of_subjects = int(input("How many subjects do they offer?\n"))

    if number_of_subjects <= 0:
        print("Invalid input")

print("Saving >>>>>>>>>>>>>>>>>>>>>>>>>")
print("Saved successfully\n")

for student_index in range(number_of_students):

    student_result = []


    for subject_index in range(number_of_subjects):

        print(f"Entering score for student {student_index + 1}")

        score = int(input(f"Enter score for subject {subject_index + 1}: "))

        while score < 0 or score > 100:

            print("Invalid score!!!")

            score = int(input(f"Enter score for subject {subject_index + 1}: "))

        student_result.append(score)

        print("Saving >>>>>>>>>>>>>>>>>>>>>>>>>")
        print("Saved successfully\n")

    student_quiz_results.append(student_result)

set_student_quiz_results(student_quiz_results)

display_student_result_table()

print()

display_subject_summary()

print()

display_class_summary()

File: StudentGrade/java/StudentGradeMain.java
import java.util.Scanner;

public class StudentGradeMain {

    public static void main(String[] args) {

        Scanner inputCollector = new Scanner(System.in);

        System.out.println("How many students do you have?");
        int numberOfStudents;
        do {
            numberOfStudents = inputCollector.nextInt();

            if (numberOfStudents <= 0) {
                System.out.println("Number of students can not be zero or negative!!!");
                System.out.println("How many students do you have?");
            }
        } while (numberOfStudents <= 0);

        System.out.println("How many subjects do they offer?");
        int numberOfSubjects;
        do {
            numberOfSubjects = inputCollector.nextInt();

            if (numberOfSubjects <= 0) {
                System.out.println("Number of quiz taken can not be zero or negative!!!");
                System.out.println("How many subjects do they offer?");
            }
        } while (numberOfSubjects <= 0);

        System.out.println("Saving >>>>>>>>>>>>>>>>>>>>>");
        System.out.println("Saved successfully\n");

        int[][] studentsResults = new int[numberOfStudents][numberOfSubjects];

        int studentIndex = 0;

        while (studentIndex < numberOfStudents) {

            for (int subjectIndex = 0; subjectIndex < numberOfSubjects; subjectIndex++) {

                System.out.println("Entering score for student " + (studentIndex + 1));
                System.out.println("Enter score for subject " + (subjectIndex + 1));
                int score;

                do {
                    score = inputCollector.nextInt();

                    if (score < 0 || score > 100) {
                        System.out.println("Invalid score!!!");
                        System.out.println("Enter score for subject " + (subjectIndex + 1));
                    }
                } while (score < 0 || score > 100);

                studentsResults[studentIndex][subjectIndex] = score;

                System.out.println("Saving >>>>>>>>>>>>>>>>>>>>>");
                System.out.println("Saved successfully\n");
            }

            studentIndex++;
        }

        StudentGrade studentGrade = new StudentGrade(studentsResults);

        studentGrade.displayStudentScoreTable();

        studentGrade.displaySubjectSummary();

        studentGrade.displayClassSummary();
    }
}